src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseAutoValidateAntiforgeryTokenTests.cs

```diff
@@ -0,0 +1,1369 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Test.Utilities.MinimalImplementations;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class UseAutoValidateAntiforgeryTokenTests : DiagnosticAnalyzerTestBase
+    {
+        protected void VerifyCSharpWithDependencies(string source, params DiagnosticResult[] expected)
+        {
+            this.VerifyCSharp(
+                new[] { source, ASPNetCoreApis.CSharp }.ToFileAndSource(),
+                expected);
+        }
+
+        [Fact]
+        public void Test_GlobalAntiForgeryFilter_Add_ChildrenOfIAsyncAuthorizationFilter_NotCallMethodsOf_DescedantOfIAntiForgery_Diagnostic()
+        {
+            VerifyCSharpWithDependencies(@"
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+[MyValidateAntiForgeryAttribute]
+class MakeSureValidateAntiForgeryAttributeIsUsedSomeWhereClass : ControllerBase
+{
+}
+
+class FilterClass : IAsyncAuthorizationFilter 
+{
+    public DefaultAntiforgery defaultAntiforgery;
+
+    public Task OnAuthorizationAsync (AuthorizationFilterContext context)
+    {
+        return null;
+    }
+}
+
+class TestClass : ControllerBase
+{
+    [HttpDelete]
+    public override AcceptedAtActionResult AcceptedAtAction (string actionName)
```

Comment:
>AcceptedAtAction [](start = 43, length = 16)

Are these controller action methods, that accepts requests routed to them?  This seems to be part of ControllerBase.

Reply:
In ControllerBase, the AcceptAtAction() has a [NonAction] attribute on it.  We shouldn't look at methods with [NonAction] on them in this analyzer.